### ISSUE TITLE
Irrelevant error occurs when data service batch request resource path is incorrect

### ISSUE THREAD

**Issue by Manuri** [wso2/product-ei#2372]
**Description:**
$subject.
Following is the error.

```bash
[2018-07-03 13:24:31,195] [EI-Core] ERROR - ServerWorker Error processing POST request for : /services/JSONSample/employee. Error detail: Value type miss match, Expected value type - 'null', but found - 'NUMBER'. 
java.lang.IllegalArgumentException: Value type miss match, Expected value type - 'null', but found - 'NUMBER'
	at org.apache.axis2.json.gson.GsonXMLStreamReader.nextValue(GsonXMLStreamReader.java:739)
	at org.apache.axis2.json.gson.GsonXMLStreamReader.readValue(GsonXMLStreamReader.java:626)
	at org.apache.axis2.json.gson.GsonXMLStreamReader.stateTransition(GsonXMLStreamReader.java:532)
	at org.apache.axis2.json.gson.GsonXMLStreamReader.next(GsonXMLStreamReader.java:178)
	at org.apache.axiom.om.impl.builder.StAXOMBuilder.parserNext(StAXOMBuilder.java:681)
	at org.apache.axiom.om.impl.builder.StAXOMBuilder.next(StAXOMBuilder.java:214)
	at org.apache.axiom.om.impl.llom.OMSerializableImpl.build(OMSerializableImpl.java:78)
	at org.apache.axiom.om.impl.llom.OMElementImpl.build(OMElementImpl.java:722)
	at org.apache.axiom.om.impl.llom.OMElementImpl.detach(OMElementImpl.java:700)
	at org.apache.axiom.om.impl.llom.OMNodeImpl.setParent(OMNodeImpl.java:105)
	at org.apache.axiom.om.impl.llom.OMElementImpl.addChild(OMElementImpl.java:296)
	at org.apache.axiom.om.impl.llom.OMElementImpl.addChild(OMElementImpl.java:212)
	at org.apache.axiom.soap.impl.llom.SOAPBodyImpl.addChild(SOAPBodyImpl.java:231)
	at org.apache.axis2.json.gson.JSONMessageHandler.invoke(JSONMessageHandler.java:84)
	at org.apache.axis2.engine.Phase.invokeHandler(Phase.java:340)
	at org.apache.axis2.engine.Phase.invoke(Phase.java:313)
	at org.apache.axis2.engine.AxisEngine.invoke(AxisEngine.java:261)
	at org.apache.axis2.engine.AxisEngine.receive(AxisEngine.java:167)
	at org.apache.synapse.transport.passthru.ServerWorker.processNonEntityEnclosingRESTHandler(ServerWorker.java:337)
	at org.apache.synapse.transport.passthru.ServerWorker.processEntityEnclosingRequest(ServerWorker.java:383)
	at org.apache.synapse.transport.passthru.ServerWorker.run(ServerWorker.java:151)
	at org.apache.axis2.transport.base.threads.NativeWorkerPool$1.run(NativeWorkerPool.java:172)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

```

**Affected Product Version:**
EI 6.3.0, EI 6.2.0

**Steps to reproduce:**
1. Deploy the JSON sample dbs in EI and enable batch requests
2. Send the following request

```bash
curl -v http://localhost:8280/services/JSONSample/employee -d @jsonsampledata.json -H'Content-Type:application/json' -X POST
```
The request works when the resource path is "employee_batch_req". The error does not imply the actual issue. Also, AFAIK whether the request is a batch request or not is derived from the payload, therefore it appears that another issue is here. 

Content of jsonsampledata.json
```json
{
    "_postemployee_batch_req": {
        "_postemployee": [
            {
                "employeeNumber": 5012,
                "lastName": "Smith",
                "firstName": "Will",
                "email": "will@smith.com",
                "salary": 13500.0
            },
            {
                "employeeNumber": 5013,
                "lastName": "Peter",
                "firstName": "Parker",
                "email": "peter@parker.com",
                "salary": 15500.0
            }
        ]
    }
}

```

**Related Issues:**
https://wso2.org/jira/browse/DS-1190 has the same error, but this issue is a resolved one. Scenario is also different.

**Comment by nazakahar** [comment]
Hi @Manuri did you manage to solve your problem? I know 2 years have since passed. Just trying my luck here.

**Comment by sidharthYenlo** [comment]
This issue is still not fixed in EI 660 WUM version : 1615822296649
